src/tests/conformance/post_handshake/validators.rs

```diff
@@ -239,48 +230,40 @@ async fn c026_TM_VALIDATOR_LIST_send_validator_list() {
         "invalid signing public key length: {}",
         master_public_bytes.len()
     );
-    let signable_manifest =
-        create_signable_manifest(1, &master_public_bytes, &signing_public_bytes);
+    let manifest = create_manifest(1, &master_public_bytes, &signing_public_bytes);
 
     // 3. append manifest prefix
-    let mut prefixed_signable = BytesMut::with_capacity(1024);
-    prefixed_signable.put(&b"MAN\x00"[..]);
-    prefixed_signable.extend_from_slice(&signable_manifest);
-
-    // 4. Sign the signable manifest with master secret key, get master signature
-    let master_signature_bytes = sign_buffer(&master_secret_key, &prefixed_signable);
-
-    // 5. Sign it with signing private key, get signature
-    let signature_bytes = sign_buffer(&signing_secret_key, &prefixed_signable);
-
-    // 6. Create final manifest with sequence, public key, signing public key, master signature, signature
-    let manifest = create_final_manifest(
-        1,
-        &master_public_bytes,
-        &signing_public_bytes,
-        &master_signature_bytes,
-        &signature_bytes,
-    );
+    //let mut prefixed_signable = BytesMut::with_capacity(1024);
+
+    // 3. Sign the signable manifest with master secret key, get master signature
+    let master_signature_bytes =
+        sign_buffer_with_prefix(MANIFEST_PREFIX, &master_secret_key, &manifest);
```

Comment:
I still think we can avoid this extra function which handles the `prefix` case

```suggestion
    let master_signature_bytes =
        sign_buffer(Some(MANIFEST_PREFIX), &master_secret_key, &manifest);
```

Then on the line below, we can reuse the function like this:
```rust
    // 8.  Get signature for blob using master private key
    let signature = sign_buffer(None, &signing_secret_key, &blob_bytes);
```

Reply:
in my opinion.
- I think it's good to have a function that just does the signing -- I don't consider it 'extra'
- appending of a prefix is somewhat orthogonal to the signing.
- I don't like passing the idea of passing in 'None'.